opentelemetry-prometheus/src/lib.rs

```diff
@@ -122,6 +135,30 @@ pub struct ExporterBuilder {
     ///
     /// If not set a new empty `Registry` is created.
     registry: Option<prometheus::Registry>,
+
+    /// The host used by the prometheus exporter
+    ///
+    /// If not set it will be defaulted to all addresses "0.0.0.0"
+    host: Option<String>,
+
+    /// The port used by the prometheus exporter
+    ///
+    /// If not set it will be defaulted to port "9464"
+    port: Option<String>,
```

Comment:
I wonder if it will make sense to use an unsigned int here?

Reply:
Yeah, I think that makes sense. I did a quick look around to see how the various frameworks and standard library handle it.

`std::net::SocketAddr` treats port as `u16` - https://doc.rust-lang.org/std/net/enum.SocketAddr.html#method.port
`std::net::ToSocketAddr` trait also shows most examples treating it as `u16` - https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html

If I look at a few Web Frameworks they are `u16` becuase they ultimately accept the SocketAddr.
- https://actix.rs/ - leverages a tuple with the port treated like a u16 for conversion to SocketAdd
- https://rocket.rs/v0.5-rc/guide/configuration/#overview
- https://github.com/hyperium/tonic/blob/master/tonic/src/transport/server/mod.rs#L622
- https://docs.rs/hyper/0.14.11/hyper/server/index.html

I can make the switch but I guess the only question is that when parsing the env variable in the impl for `default` how would we handle the potential parse error? I guess it would just get ignored and we fall back to default port? The only problem with that is we are not giving the user any feedback that it occurred so would panic early be a better dev experience?

Reply:
That's a good question. I think a good approach is to fall back to default one and print a error message using `global::handle_error` function

Reply:
Cool, I have gone ahead and made the change and logged an error message.